libyul/optimiser/BlockHasher.h

```diff
@@ -27,6 +27,38 @@
 namespace solidity::yul
 {
 
+class HasherBase
```

Comment:
I know you've only just extracted this into a base class, but it would be nice if this hashing magic was documented a bit - if nothing, at least the `fnvPrime` and `fnvEmptyHash` values?

Reply:
Right, but I still don't know why it's used over the default hashing algorithm. In any case, this isn't particularly important, especially seeing as @ekpyron will be off until Monday, and this certainly shouldn't be a merge blocker.

Reply:
For the record: FNV-1 was originally just a simple placeholder here - I actually tried replacing it with something more sophisticated and technically faster and better avalanching like xxhash64 once, but it made no difference at all, so it just stayed with FNV-1.

Reply:
What do you mean with "default hashing algorithm"? You mean like ``std::hash``? Can't use that - that's implementation dependent and this needs to be universally deterministic and reproducible. Or ``keccak``? Too slow for this purpose and we don't need anything cryptographic here.